pygac/pod_reader.py

```diff
@@ -332,20 +333,22 @@ def read_header(cls, filename, fileobj=None):
                 fd_.seek(0)
                 tbm_head = None
                 tbm_offset = 0
-            # read header
-            head0, = np.frombuffer(
-                fd_.read(header0.itemsize),
-                dtype=header0, count=1)
-            year, jday, _ = cls.decode_timestamps(head0["start_time"])
-            start_date = (datetime.date(year, 1, 1) +
-                          datetime.timedelta(days=int(jday) - 1))
-            if start_date < datetime.date(1992, 9, 8):
-                header = header1
-            elif start_date <= datetime.date(1994, 11, 15):
-                header = header2
-            else:
-                header = header3
-            fd_.seek(tbm_offset, 0)
+            header = header3
+            if not eosip_header:
+                # choose the appropriate header
+                head0, = np.frombuffer(
+                    fd_.read(header0.itemsize),
+                    dtype=header0, count=1)
+                year, jday, _ = cls.decode_timestamps(head0["start_time"])
+                start_date = (datetime.date(year, 1, 1) +
+                              datetime.timedelta(days=int(jday) - 1))
+                if start_date < datetime.date(1992, 9, 8):
+                    header = header1
+                elif start_date <= datetime.date(1994, 11, 15):
+                    header = header2
+                else:
+                    header = header3
+                fd_.seek(tbm_offset, 0)
```

Comment:
This could be extracted to a method `choose_header_based_on_timestamp`

Reply:
Done